tests/001-handshake.cpp

```diff
@@ -108,68 +109,116 @@ namespace oxen::quic::test
+
+            auto [local_tls, _] = defaults::tls_creds_from_ed_keys();
 
             auto ep_notls = test_net.endpoint(default_addr);
             auto ep_tls = test_net.endpoint(local_addr);
 
-            // ep_notls->listen();  // Shouldn't compile if uncommented!
-            // ep_notls->listen(local_tls, local_tls);  // Nor this
             REQUIRE_NOTHROW(ep_tls->listen(local_tls));
         };
+    };
 
-        SECTION("Endpoint::listen() + Endpoint::Connect() - Default addressing")
-        {
-            Network test_net{};
-            opt::local_addr default_addr{};
-            auto local_tls = GNUTLSCreds::make("./serverkey.pem"s, "./servercert.pem"s, "./clientcert.pem"s);
+    TEST_CASE("001 - Handshaking: Client Validation", "[001][client]")
+    {
+        auto client_established = callback_waiter{[](connection_interface&) {}};
+        auto server_established = callback_waiter{[](connection_interface&) {}};
 
-            auto ep = test_net.endpoint(default_addr);
+        Network test_net{};
 
-            REQUIRE_NOTHROW(ep->listen(local_tls));
-            REQUIRE_THROWS(ep->connect(default_addr, local_tls));
-        };
+        auto [client_tls, server_tls] = defaults::tls_creds_from_ed_keys();
 
-        SECTION("Endpoint::connect() - Immediate network shutdown")
-        {
-            Network test_net{};
-            test_net.set_shutdown_immediate();
-            auto server_tls = GNUTLSCreds::make("./serverkey.pem"s, "./servercert.pem"s, "./clientcert.pem"s);
-            auto client_tls = GNUTLSCreds::make("./clientkey.pem"s, "./clientcert.pem"s, "./servercert.pem"s);
+        opt::local_addr server_local{};
+        opt::local_addr client_local{};
 
-            opt::local_addr server_local{};
-            opt::local_addr client_local{};
+        auto server_endpoint = test_net.endpoint(server_local, server_established);
+        REQUIRE(server_endpoint->listen(server_tls));
 
-            auto server_endpoint = test_net.endpoint(server_local);
-            REQUIRE(server_endpoint->listen(server_tls));
+        SECTION("Endpoint::listen() + Endpoint::Connect() - Incorrect pubkey in remote")
+        {
+            uint64_t client_error{0};
 
-            opt::local_addr client_remote{"127.0.0.1"s, server_endpoint->local().port()};
+            auto client_closed = callback_waiter{[&client_error](connection_interface&, uint64_t) { client_error = 1000; }};
+            opt::remote_addr client_remote{defaults::CLIENT_PUBKEY, "127.0.0.1"s, server_endpoint->local().port()};
 
-            auto client_endpoint = test_net.endpoint(client_local);
-            REQUIRE_NOTHROW(client_endpoint->connect(client_remote, client_tls));
+            auto client_endpoint = test_net.endpoint(client_local, client_established, client_closed);
+            auto client_ci = client_endpoint->connect(client_remote, client_tls);
+
+            REQUIRE(not client_established.wait());
```

Comment:
Shouldn't this be a `REQUIRE(client_closed.wait())` since that's what we expect to trigger in this case?